### PR TITLE
Commit generated Go code and bump dependencies

### PR DESCRIPTION
**Pull request by pellared**
# Changes

I did the most important changes (from my point of view) to make the Go code easier to maintain:

1. Commit gRPC related generated code ([in Go it is idiomatic to commit the generated code](https://stackoverflow.com/q/56415527/859275), also without it it is not easy to bump dependencies)
2. Change a little bit the way the code is generated - use https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module + `go:generate`. This is more idiomatic.
4. Bump the dependencies. Fixes https://github.com/open-telemetry/opentelemetry-demo/issues/923

## Merge Requirements

For new features contributions please make sure you have completed the following
essential items:

* [x] ~~`CHANGELOG.md` updated to document new feature additions~~
* [x] ~~Appropriate documentation updates in the [docs][]~~
* [x] ~~Appropriate Helm chart updates in the [helm-charts][]~~

Maintainers will not merge until the above have been completed. If you're unsure
which docs need to be changed ping the
[@open-telemetry/demo-approvers](https://github.com/orgs/open-telemetry/teams/demo-approvers).

[docs]: https://opentelemetry.io/docs/demo/
[helm-charts]: https://github.com/open-telemetry/opentelemetry-helm-charts

### CODE REVIEW

File: src/accountingservice/README.md [hunk 1]
```diff
@@ -18,3 +17,23 @@ From the root directory, run:
...
+## Regenerate protos
+
+> **Note**
+> [`protoc`](https://grpc.io/docs/protoc-installation/) is required.
+
+To regenerate gRPC code run:
+
+```sh
+go generate
+```
```
Comment (pellared) [inline src/accountingservice/README.md:30]: Do we want to add a job to the "Checks" GitHub workflow that would make sure that the committed generated code is up-to-date? Do we want to do this in this same PR?

**Comment by pellared** [comment]
@open-telemetry/go-approvers PTAL

**Review by dashpole — APPROVED** [review]
LGTM!

**Review by cartersocha — APPROVED** [review]
Lgtm

File: src/accountingservice/README.md (hunk 1 above)
Comment (cartersocha) [reply src/accountingservice/README.md:30]: Let’s do a separate PR if you wouldn’t mind at least creating an issue for it. 

I’ll go ahead and merge this

**Pull request merged**